lib/http/HttpClient_WinInet.cpp

```diff
@@ -136,6 +139,105 @@ class WinInetRequestWrapper
         return true;
     }
 
+    DWORD GetDWORDRegKey(HKEY hKey, const std::string& subKey, const std::string& value) {
+        DWORD data;
+        DWORD dataSize = sizeof(data);
+        LONG res = RegGetValueA(hKey, subKey.c_str(), value.c_str(), RRF_RT_REG_DWORD, NULL, &data, 
+            &dataSize);
+        if (res != ERROR_SUCCESS) {
+            return 0;
+        }
+        return data;
+    }
+
+    std::string GetStringRegKey(HKEY hKey, const std::string& subKey, const std::string& value) {
+        DWORD stringSize;
+        LONG getSizeResult = RegGetValueA(hKey, subKey.c_str(), value.c_str(), RRF_RT_REG_SZ, NULL, 
+            NULL, &stringSize);
+        if (getSizeResult != ERROR_SUCCESS) {
+            return "";
```

Comment:
use the default string constructor, rather than the c-string constructor:
`std::string{}`